addons/iap_mail/static/src/js/services/iap_notification_service.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-module **/
+
+import { Markup } from 'web.utils';
+import { registry } from "@web/core/registry";
+
+export const iapNotificationService = {
+    dependencies: ["notification"],
+
+    start(env, { notification }) {
+        env.bus.on("WEB_CLIENT_READY", null, async () => {
+            const legacyEnv = owl.Component.env;
+            legacyEnv.services.bus_service.onNotification(this, (notifications) => {
+                for (const { payload, type } of notifications) {
+                    if (type === 'iap_notification') {
+                        displayIapNotification(payload);
+                    }
+                }
+            });
+            legacyEnv.services.bus_service.startPolling();
+        });
+
+        /**
+         * Displays the IAP notification on user's screen
+         */
+        function displayIapNotification(notif) {
+            if (notif.url) {
+                const message = Markup`<a class='btn btn-link' href=${notif.url} target='_blank' ><i class='fa fa-arrow-right'></i> ${env._t("Buy more credits")}</a>`;
```

Comment:
As previously mentioned, URL should be escaped (as html attribute) as it is injected into code. Also isn't href missing quotes?

Reply:
And as I also mentioned before, I would also escape the translated part, even if we usually don't care about translation injection in terms of security, but even just if the translator decided to put a < in the string for some reason it would no longer be a valid DOM if it's not escaped.